sentry-contexts/src/utils.rs

```diff
@@ -41,6 +48,18 @@ mod model_support {
         }
     }
 
+    pub fn get_model() -> Option<String> {
+        sysctlbyname_call("hw.model")
+    }
+
+    pub fn get_macos_version() -> Option<String> {
+        sysctlbyname_call("kern.osproductversion")
```

Comment:
`kern.osproductversion` seems to have been introduced in [xnu-4570.51.1](https://github.com/apple/darwin-xnu/commit/5bbb823c13f3ab1ab58878f96b35433a29882676#diff-1daf70413d68262120e719485aac47dd7da773d2333383c9fe5fd34dc93a5ffc), so it seems to be supported in macOS 10.13.4 and later (per [opensource.apple.com](https://opensource.apple.com/releases/)). What should the SDK do if `kern.osproductversion` is not available? Should it return `None` or use a heuristic to determine the version?

Reply:
I think returning `None` is a good idea. Luckily macOS has very quick deprecation/EOL cycles, so you don’t need to worry about older versions for too long.